x-pack/legacy/plugins/monitoring/server/init_monitoring_xpack_info.js

```diff
@@ -7,15 +7,26 @@
 import { checkLicenseGenerator } from './cluster_alerts/check_license';
 import { hasMonitoringCluster } from './es_client/instantiate_client';
 import { LOGGING_TAG } from '../common/constants';
+import { XPackInfo } from '../../xpack_main/server/lib/xpack_info';
 
 /*
  * Expose xpackInfo for the Monitoring cluster as server.plugins.monitoring.info
  */
-export const initMonitoringXpackInfo = async ({ config, xpackMainPlugin, expose, log }) => {
+export const initMonitoringXpackInfo = async ({
+  config,
+  server,
+  client,
+  xpackMainPlugin,
+  licensing,
+  expose,
+  log,
+}) => {
   const xpackInfo = hasMonitoringCluster(config)
-    ? xpackMainPlugin.createXPackInfo({
```

Comment:
Would you mind removing `createXPackInfo` from x-pack? There are no other consumers in our codebase.